daemon/info.go

```diff
@@ -192,15 +118,100 @@ func (daemon *Daemon) SystemVersion() types.Version {
 	return v
 }
 
-func (daemon *Daemon) showPluginsInfo() types.PluginsInfo {
-	var pluginsInfo types.PluginsInfo
+func (daemon *Daemon) fillDriverInfo(v *types.Info) {
+	var ds [][2]string
+	drivers := ""
+	statuses := daemon.imageService.LayerStoreStatus()
+	for os, gd := range daemon.graphDrivers {
+		ds = append(ds, statuses[os]...)
+		drivers += gd
+		if len(daemon.graphDrivers) > 1 {
+			drivers += fmt.Sprintf(" (%s) ", os)
+		}
+	}
+	drivers = strings.TrimSpace(drivers)
 
-	pluginsInfo.Volume = daemon.volumes.GetDriverList()
-	pluginsInfo.Network = daemon.GetNetworkDriverList()
-	// The authorization plugins are returned in the order they are
-	// used as they constitute a request/response modification chain.
-	pluginsInfo.Authorization = daemon.configStore.AuthorizationPlugins
-	pluginsInfo.Log = logger.ListDrivers()
+	v.Driver = drivers
+	v.DriverStatus = ds
+}
 
-	return pluginsInfo
+func (daemon *Daemon) fillPluginsInfo(v *types.Info) {
+	v.Plugins = types.PluginsInfo{
+		Volume:  daemon.volumes.GetDriverList(),
+		Network: daemon.GetNetworkDriverList(),
+
+		// The authorization plugins are returned in the order they are
+		// used as they constitute a request/response modification chain.
+		Authorization: daemon.configStore.AuthorizationPlugins,
+		Log:           logger.ListDrivers(),
+	}
+}
+
+func (daemon *Daemon) fillSecurityOptions(v *types.Info, sysInfo *sysinfo.SysInfo) {
+	var securityOptions []string
+	if sysInfo.AppArmor {
+		securityOptions = append(securityOptions, "name=apparmor")
+	}
+	if sysInfo.Seccomp && supportsSeccomp {
+		profile := daemon.seccompProfilePath
+		if profile == "" {
+			profile = "default"
+		}
+		securityOptions = append(securityOptions, fmt.Sprintf("name=seccomp,profile=%s", profile))
+	}
+	if selinuxEnabled() {
+		securityOptions = append(securityOptions, "name=selinux")
+	}
+	if rootIDs := daemon.idMappings.RootPair(); rootIDs.UID != 0 || rootIDs.GID != 0 {
+		securityOptions = append(securityOptions, "name=userns")
+	}
+	v.SecurityOptions = securityOptions
+}
+
+func hostName() string {
+	hostname := ""
+	if hn, err := os.Hostname(); err != nil {
+		logrus.Warnf("Could not get hostname: %v", err)
+	} else {
+		hostname = hn
+	}
+	return hostname
+}
+
+func kernelVersion() string {
+	kernelVersion := "<unknown>"
```

Comment:
FWIW; I don't like these `<unknown>` values; they should be in the CLI, not in the API, but that would need some discussion (for backward compatibility)